### DISCUSSION TITLE
wasm file downloads twice

### DISCUSSION THREAD

**Discussion by MarkZamoyta** [jeffreylanters/react-unity-webgl#580] [Help Wanted!]
### Please avoid duplicates

- [X] I checked [all open issues](https://github.com/jeffreylanters/react-unity-webgl/issues) and [discussions](https://github.com/jeffreylanters/react-unity-webgl/discussions) and none of them matched my problem.

### Language and Compiler

TypeScript's built-in Compiler

### What environment are you using?

Local Development Server

### When does your problem occur?

When the Unity Component mounts

### What does your problem relate to?

The problem seems Unity related

### React-Unity-WebGL Version

9.4.3

### React Version

17.0.2

### Unity Version

2021.3.16f1

### What happened?

My game downloads but gets stuck for 25 seconds at 90%.  I looked at what was going on during that time, and the browser is downloading the wasm file twice.  On a chromebook with limited resources, it doesn't get cached (ever), so this doubles the download time.  On a PC, everything is fine because the wasm file is cached after the first download, and the 2nd download is near instant.
Is there a way to prevent it being downloaded twice?  It happens somewhere in useUnityContext() before it sets isLoaded to true.

### Would you be interested in contributing a fix?

- [ ] yes, I would like to contribute a fix

**Comment by jeffreylanters** [comment]
would it be able for you to share some code? it might be caused by unintentional rerenders

**Comment by lzqlawrence** [comment]
i also got the same issue on my PC chrome and mobile chrome. the *.wasm file will download twice.

**Comment by MarkZamoyta** [comment]
I looked into rerenders, I'm new to react so I figured that could be the
case, but rerendering wasn't the problem.
Unfortunately many schools have extremely slow wifi, so I was forced to
drop Unity for just plain JS, which is about 1/20th the size for my game.
Even downloading a Unity game once is way too slow, and Chrome won't cache
it on limited devices.
What's the best way to get you my code in case it can help someone else?
Thanks,
Mark

**Comment by lzqlawrence** [comment]
> i also got the same issue on my PC chrome and mobile chrome. the *.wasm file will download twice.

when i build a release build, it's fine. so the *.wasm file only download twice in development build

**Comment by lia12345kimo** [comment]
I found the reason
web server settgin mine type

looking 
https://docs.unity3d.com/2021.2/Documentation/Manual/webgl-server-configuration-code-samples.html

file type ==> .wasm
mimeType ==> application/wasm

You can avoid it twice
